test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/Metadata/ModelMetadataProviderTest.cs

```diff
@@ -421,6 +423,60 @@ public void DisplayAttribute_Description()
         }
 
         [Fact]
+        public void DisplayName_FromResources_GetsRecomputed()
+        {
+            // Arrange
+            var display = new DisplayAttribute()
+            {
+                Name = nameof(TestResources.DisplayAttribute_Name2),
+                ResourceType = typeof(TestResources),
+            };
+
+            var provider = CreateProvider(new[] { display });
+            var metadata = provider.GetMetadataForType(typeof(string));
+            var currentCulture = Thread.CurrentThread.CurrentUICulture;
+
+            // Act & Assert
+            var result = metadata.DisplayName;
+            Assert.Equal("name from resources" + currentCulture.Name, result);
```

Comment:
This is not at all reliable. Never depend on the ambient locale in a test unless within a context protected by `[ReplaceCulture]`, `CultureReplacer`, or `CultureReplacingMiddleware`.

Suggest changing this `// Act & Assert` block to loop through two or three UI culture names and consistently use `CultureReplacer` in that loop. Same for the new test below.